torch/nested/_internal/nested_tensor.py

```diff
@@ -198,6 +200,20 @@ def __tensor_unflatten__(inner_tensors: Dict, meta, outer_size, outer_stride):
             # Associate offsets or lengths (possibly fake, possibly functionalized)
             # with the ragged_size.
             ragged_size = outer_size[ragged_idx]
+            # Ordinarily, we create singleton ints by passing a 1D tensor to
+            # get_tensor_symint. When we create singleton ints this way, we are
+            # able to set the singleton int's metadata, e.g. the 'vec' field
+            # during construction.
+            #
+            # In the case of torch compile, a symbolic version of the
+            # singleton int does not yet have the metadata set, and so we set it
+            # here.
+            #
+            # The invariant today is that we must guarantee that singleton ints
+            # acquire its metadata by the time it finds its way onto some
+            # NestedTensor.
+            ragged_size.node._singleton_vec = offsets
+            ragged_size.node._singleton_sum_vec = values.shape[0]
```

Comment:
???

Reply:
This should be done when we are creating the SymNode/symbol initially (i.e. during fakeification). You should be able to read it off when you are fakeifying. There is some annoyance which is that maybe the offsets tensor hasn't been fakeified yet

Reply:
Updated to use the recursive fakification idea as discussed offline.